api/base/generic_bulk_views.py

```diff
@@ -0,0 +1,110 @@
+
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework_bulk import generics as bulk_generics
+from rest_framework.exceptions import PermissionDenied, ValidationError
+
+from framework.auth.core import Auth
+
+from website.project.model import Q
+from api.base.settings import BULK_SETTINGS
+from api.base.exceptions import Conflict, JSONAPIException
+from api.base.utils import is_bulk_request
+
+
+class ListBulkCreateJSONAPIView(bulk_generics.ListBulkCreateAPIView):
+    """
+    Custom ListBulkCreateAPIView that properly formats bulk create responses
+    in accordance with the JSON API spec
+    """
+
+    # overrides ListBulkCreateAPIView
+    def create(self, request, *args, **kwargs):
+        """
+        Correctly formats both bulk and single POST response
+        """
+        if is_bulk_request(request):
+            if not request.data:
+                raise ValidationError('Request must contain array of resource identifier objects.')
+
+        response = super(ListBulkCreateJSONAPIView, self).create(request, *args, **kwargs)
+        if 'data' in response.data:
+            return response
+        return Response({'data': response.data}, status=status.HTTP_201_CREATED)
```

Comment:
Rather than returning a new `Response`, could you format and set `response's` data? Consider the following use case:

```python
class MyListBulkView(ListBulkCreateJSONAPIView):

    def create(self, ...):
        if some_condition:
            return Response(..., status=status.HTTP_301_MOVED_PERMANENTLY)
```

With the current code, the status code would incorrectly be overwritten.